libraries/inverse/hpiFit/sensorset.cpp

```diff
@@ -0,0 +1,151 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    SensorSet class definition.
+ *
+ */
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include <inverse/hpiFit/sensorset.h>
+#include <iostream>
+#include <fwd/fwd_coil_set.h>
+#include <fiff/fiff_ch_info.h>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QCoreApplication>
+
+//=============================================================================================================
+// EIGEN INCLUDES
+//=============================================================================================================
+
+#include <Eigen/Dense>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace INVERSELIB;
+using namespace FIFFLIB;
+using namespace FWDLIB;
+using namespace Eigen;
+
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+SensorSet::SensorSet(const FwdCoilSet::SPtr pFwdSensorSet)
+{
+    if(pFwdSensorSet!=nullptr) {
+        convertFromFwdCoilSet(pFwdSensorSet);
+    }
+}
+
+//=============================================================================================================
+
+void SensorSet::convertFromFwdCoilSet(const QSharedPointer<FWDLIB::FwdCoilSet> pFwdSensorSet)
```

Comment:
Can we refactor this to not be a shared pointer but const ref instead?

Reply:
Also the naming of the function might be confusing. How about `initFromFwdCoilSet`?

Reply:
> Can we refactor this to not be a shared pointer but const ref instead?

All members of FwdCoilSet return only pointers. So this would require a refactor of FwdCoilSet and not only my code.